src/services/TestResultsService.ts

```diff
@@ -195,7 +195,7 @@ export class TestResultsService {
 
     payload = this.setCreatedAtAndLastUpdatedAtDates(payload);
     return this.getTestTypesWithTestCodesAndClassification(payload.testTypes, payload.vehicleType, payload.vehicleSize, payload.vehicleConfiguration,
-        payload.noOfAxles, payload.euVehicleCategory, payload.vehicleClass.code, payload.vehicleSubclass ? payload.vehicleSubclass[0] : undefined, payload.numberOfWheelsDriven)
+        payload.noOfAxles, payload.euVehicleCategory, payload.vehicleClass ? payload.vehicleClass.code : undefined, payload.vehicleSubclass ? payload.vehicleSubclass[0] : undefined, payload.numberOfWheelsDriven)
```

Comment:
We could have used && operators instead of ternary here to make it more concise I think

Reply:
As discussed, I upgraded typescript version to 3.8.3 and used optional chaining